src/transaction/TransactionController.ts

```diff
@@ -1212,18 +1238,175 @@ export class TransactionController extends BaseController<
       status === TransactionStatus.confirmed ||
       status === TransactionStatus.failed ||
       status === TransactionStatus.cancelled
     );
   }
+
+  private etherscanTransactionStateReconciler(
+    remoteTxs: TransactionMeta[],
+    localTxs: TransactionMeta[],
+  ): [boolean, TransactionMeta[]] {
+    const outdatedTxs: TransactionMeta[] = this.getOutdatedTransactions(
+      remoteTxs,
+      localTxs,
+    );
+
+    const newTxs: TransactionMeta[] = this.getNewTransactions(
+      remoteTxs,
+      localTxs,
+    );
+
+    const updatedLocalTxs = localTxs.map((tx: TransactionMeta) => {
+      const txIdx = outdatedTxs.findIndex(
+        ({ transactionHash }) => transactionHash === tx.transactionHash,
+      );
+      return txIdx === -1 ? tx : outdatedTxs[txIdx];
+    });
+
+    const updateRequired = newTxs.length > 0 || updatedLocalTxs.length > 0;
+
+    return [updateRequired, [...newTxs, ...updatedLocalTxs]];
+  }
+
+  /**
+   * Get all transactions that are in the remote transactions array
+   * but not in the local transactions array
+   * @param remoteTxs - Array of transactions from remote source
+   * @param localTxs - Array of transactions stored locally
+   * @returns TransactionMeta array
+   */
+  private getNewTransactions(
+    remoteTxs: TransactionMeta[],
+    localTxs: TransactionMeta[],
+  ): TransactionMeta[] {
+    return remoteTxs.filter((tx) => {
+      const alreadyInTransactions = localTxs.find(
+        ({ transactionHash }) => transactionHash === tx.transactionHash,
+      );
+      return !alreadyInTransactions;
+    });
+  }
+
+  /**
+   * Get all the transactions that are locally outdated with respect
+   * to a remote source (etherscan or blockchain). The returned array
+   * contains the transactions with the updated data.
+   * @param remoteTxs - Array of transactions from remote source
+   * @param localTxs - Array of transactions stored locally
+   * @returns TransactionMeta array
+   */
+  private getOutdatedTransactions(
+    remoteTxs: TransactionMeta[],
+    localTxs: TransactionMeta[],
+  ): TransactionMeta[] {
+    return remoteTxs.filter((remoteTx) => {
+      const isTxOutdated = localTxs.find((localTx) => {
+        return (
+          remoteTx.transactionHash === localTx.transactionHash &&
+          this.isTransactionOutdated(remoteTx, localTx)
+        );
+      });
+      return isTxOutdated;
+    });
+  }
+
+  /**
+   * Verifies if a local transaction is outdated in respect the remote transaction
+   * @param remoteTx - Remote transaction from Etherscan
+   * @param localTx - Local transaction
+   * @returns boolean
+   */
+  private isTransactionOutdated(
+    remoteTx: TransactionMeta,
+    localTx: TransactionMeta,
+  ): boolean {
+    const statusOutdated = this.isStatusOutdated(
+      remoteTx.transactionHash,
+      localTx.transactionHash,
+      remoteTx.status,
+      localTx.status,
+    );
+    const gasDataOutdated = this.isGasDataOutdated(
```

Comment:
These parameters should never change. They're a part of `txParams`, which is hashed to get the transaction hash. So unless I'm missing something, this should always return `false`.

Reply:
Sorry no, scratch that, `gasUsed` can change. That is unset until the transaction is confirmed, then it gets set. `gasUsed` is not in `txParams`.

Reply:
@Gudahtt do you refer to only `gas` and `gasPrice`? or also `gasUsed`?

Reply:
I hadn't noticed `gasUsed` was here at first. Only the `gas` and `gasPrice` checks are redundant here. THh `gasUsed` check is legitimate.

Though even for `gasUsed`, I would not expect that to change unless the status changes. But I do not know that for sure, so :shrug:

Reply:
I'll remove both the `gas` and `gasPrice`. I'm also not sure if the `gasUsed` will get updated at the same time as the `status`.